core/typeSugar.ml

```diff
@@ -2377,14 +2478,14 @@ let type_pattern ?(linear_vars=true) closed
              (* Construct operation type, i.e. op : A -> B or op : B *)
              match domain, codomain with
              | [], [] | _, [] -> assert false (* The continuation is at least unary *)
-             | [], [t] -> Types.Operation (Types.unit_type, t)
-             | [], ts -> Types.make_tuple_type ts
+             | [], [t] -> Types.Operation (Types.unit_type, t, is_lincase)
+             | [], ts -> Types.make_tuple_type ts  (* FIXME: WT: I don't understand this case *)
```

Comment:
Then why all other cases return an operation type, but this case returns a tuple type? Is this followed by some desugaring? The problem is I need to find somewhere to put the `is_lincase`.

Reply:
I suggest experimenting with this in a separate patch. 

Regarding `TypeUtils.arg_types`: I believe this is because we always represent the domain of function types as a record (recall functions are `n`-ary in Links). More specifically, it is a tuple, as the labels are `1` through `n`. We represent tuples as records.